src/CommunityToolkit.Maui/Views/Popup/PopupExtensions.shared.cs

```diff
@@ -43,13 +48,23 @@ public static partial class PopupExtensions
 	/// </returns>
 	public static Task<object?> ShowPopupAsync<TPopup>(this Page page, TPopup popup) where TPopup : Popup
 	{
-#if WINDOWS
-		return PlatformShowPopupAsync(popup, GetMauiContext(page));
-#else
+		if (page.IsPlatformEnabled)
+		{
+			return CreateAndShowPopupAsync(page, popup);
+		}
+		else
+		{
+			var taskCompletionSource = new TaskCompletionSource<object?>();
 
-		CreatePopup(page, popup);
-		return popup.Result;
-#endif
+			page.NavigatedTo += async (o, e) =>
+			{
+				var result = await CreateAndShowPopupAsync(page, popup);
```

Comment:
how about we use the `.ContinueWith` and not make this async? If this means a lot of extra work, we can leave this way